src/ProSidebar/Menu/SubMenu.tsx

```diff
@@ -41,7 +41,8 @@ const SubMenu: React.ForwardRefRenderFunction<unknown, Props> = (
   const referenceElement = useRef(null);
   const popperElement = useRef(null);
 
-  const handleToggleSubMenu = () => {
+  const handleToggleSubMenu = (ev) => {
+    rest.onClick && rest.onClick(ev);
```

Comment:
it may be better to add `onOpenChange`  and add it to the props Interface
and probably the function should have new submenu state (closed or opened ) something like `onOpenChange(closed)` so the user can do some actions according to it

Reply:
@azouaoui-med , fixed